tests/acceptance/course-page/code-examples/export-to-github-test.js

```diff
@@ -0,0 +1,173 @@
+import catalogPage from 'codecrafters-frontend/tests/pages/catalog-page';
+import codeExamplesPage from 'codecrafters-frontend/tests/pages/course/code-examples-page';
+import coursePage from 'codecrafters-frontend/tests/pages/course-page';
+import createCommunityCourseStageSolution from 'codecrafters-frontend/mirage/utils/create-community-course-stage-solution';
+import percySnapshot from '@percy/ember';
+import testScenario from 'codecrafters-frontend/mirage/scenarios/test';
+import { setupAnimationTest } from 'ember-animated/test-support';
+import { module, test } from 'qunit';
+import { setupApplicationTest } from 'codecrafters-frontend/tests/helpers';
+import { signIn } from 'codecrafters-frontend/tests/support/authentication-helpers';
+import { settled } from '@ember/test-helpers';
+import courseOverviewPage from 'codecrafters-frontend/tests/pages/course-overview-page';
+
+module('Acceptance | course-page | code-examples | export-to-github', function (hooks) {
+  setupApplicationTest(hooks);
+  setupAnimationTest(hooks);
+
+  async function setupTestScenario(server) {
+    testScenario(server);
+    const currentUser = signIn(this.owner, server);
+
+    const solutionAuthor = server.create('user', {
+      avatarUrl: 'https://github.com/author.png',
+      createdAt: new Date(),
+      githubUsername: 'author',
+      username: 'author',
+    });
+
+    const redis = server.schema.courses.findBy({ slug: 'redis' });
+    const python = server.schema.languages.findBy({ slug: 'python' });
+
+    server.create('repository', 'withFirstStageCompleted', {
+      course: redis,
+      language: python,
+      user: currentUser,
+    });
+
+    return {
+      currentUser,
+      solutionAuthor,
+      redis,
+      python,
+      solution: createCommunityCourseStageSolution(server, redis, 2, python, solutionAuthor),
+    };
+  }
+
+  async function navigateToCodeExamples() {
+    await catalogPage.visit();
+    await catalogPage.clickOnCourse('Build your own Redis');
+    await courseOverviewPage.clickOnStartCourse();
+    await coursePage.sidebar.clickOnStepListItem('Respond to PING');
+    await coursePage.clickOnHeaderTabLink('Code Examples');
+  }
+
+  test('creates new export when none exists', async function (assert) {
+    const { solution } = await setupTestScenario.call(this, this.server);
+
+    let exportCreateCalled = false;
+    this.server.post('/community-solution-exports', function (schema) {
+      exportCreateCalled = true;
```

Comment:
@Arpan-206 I'd like to understand your thought process here a bit more - can you find me 3 other places in the codebase where we test creating a resource, and how they handle testing that action?